### PR TITLE
chore: Switch to using `test_data_ssa_refactor` for `acir_tests`

### LINKED ISSUE #625: Update ACIR proof tests to use artifacts from Noir's new SSA
See https://github.com/noir-lang/noir/issues/2020.

Barretenberg needs to update the directory from which is it pulling the ACIR artifacts so that we can ensure test coverage through the migration.

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

Resolves #625 

This also updates.hpp file as ACIR has changed since then

# Checklist:

- [ ] I have reviewed my diff in github, line by line.
- [ ] Every change is related to the PR description.
- [ ] The branch has been merged with/rebased against the head of its merge target.
- [ ] There are no unexpected formatting changes, superfluous debug logs, or commented-out code.
- [ ] There are no circuit changes, OR a cryptographer has been assigned for review.
- [ ] New functions, classes, etc. have been documented according to the doxygen comment format. Classes and structs must have `@brief` describing the intended functionality.
- [ ] If existing code has been modified, such documentation has been added or updated.
- [ ] No superfluous `include` directives have been added.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) to any issue(s) it resolves.
- [ ] I'm happy for the PR to be merged at the reviewer's next convenience.

### CODE REVIEW

File: acir_tests/run_acir_tests.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ fail_dirs=$(grep "^fail" config.toml | sed 's/fail = \[//;s/\]//;s/\"//g;s/ //g'
 # Convert them to array
 exclude_array=($exclude_dirs)
 fail_array=($fail_dirs)
-skip_array=(diamond_deps_0 workspace workspace_default_member)
+skip_array=(diamond_deps_0 workspace workspace_default_member 9_conditional)
```
Comment (kevaundray) [inline acir_tests/run_acir_tests.sh]: 9_conditional is not passing currently which is why we have this here for now

File: acir_tests/run_acir_tests.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@ set -e
 
 BB=$PWD/${BB:-../cpp/build/bin/bb}
 CRS_PATH=~/.bb-crs
-BRANCH=master
+BRANCH=kw/add-bb-tests
```
Comment (kevaundray) [inline acir_tests/run_acir_tests.sh]: change to master once https://github.com/noir-lang/noir/pull/2023 is merged

Comment (kevaundray) [inline acir_tests/run_acir_tests.sh]:
```suggestion
BRANCH=master
```

File: acir_tests/run_acir_tests.sh (hunk 1 above)
Comment (kevaundray) [reply acir_tests/run_acir_tests.sh]: Error:

```
Testing ./9_conditional... FAILED
using cached crs at: "/mnt/user-data/kev/.bb-crs"

gzip: stdin: not in gzip format
Input is not large enough
```

Comment (kevaundray) [reply acir_tests/run_acir_tests.sh]: The problem here is that MemoryInit is not being handled -- cc @guipublic

File: acir_tests/run_acir_tests.sh [hunk 3]
```diff
@@ -7,7 +7,7 @@ set -e
 
 BB=$PWD/${BB:-../cpp/build/bin/bb}
 CRS_PATH=~/.bb-crs
-BRANCH=master
+BRANCH=kw/update-acir-tests
```
Comment (kevaundray) [inline acir_tests/run_acir_tests.sh]:
```suggestion
BRANCH=master
```

Comment (kevaundray) [reply acir_tests/run_acir_tests.sh]: Can update once https://github.com/noir-lang/noir/pull/2037 is merged

**Review by vezenovm — APPROVED** [review]

**Pull request merged**